contracts/votingMachines/GenesisProtocolLogic.sol

```diff
@@ -0,0 +1,652 @@
+        uint lostReputation;
+        if (proposal.winningVote == YES) {
+            lostReputation = proposal.preBoostedVotes[NO];
+        } else {
+            lostReputation = proposal.preBoostedVotes[YES];
+        }
+        lostReputation = (lostReputation * params.votersReputationLossRatio)/100;
+        //as staker
+        Staker storage staker = proposal.stakers[_beneficiary];
+        if (staker.amount > 0) {
+            if (proposal.state == ProposalState.Closed) {
+                //Stakes of a proposal that expires in Queue are sent back to stakers
+                rewards[0] = staker.amount;
+          }else if (staker.vote == proposal.winningVote) {
+                    uint totalWinningStakes = proposal.stakes[proposal.winningVote];
+                    uint totalStakes = proposal.stakes[YES]+proposal.stakes[NO];
+                    if (totalWinningStakes != 0) {
+                        if (staker.vote == YES) {
+                            uint _totalStakes = ((totalStakes*(100 - proposal.expirationCallBountyPercentage))/100) - proposal.daoBounty;
+                            rewards[0] = (staker.amount*_totalStakes)/totalWinningStakes;
+                        } else {
+                            rewards[0] = (staker.amount*totalStakes)/totalWinningStakes;
+                        }
+                 }
+          }
+            staker.amount = 0;
+        }
+        //as voter
+        Voter storage voter = proposal.voters[_beneficiary];
+        if ((voter.reputation != 0 ) && (voter.preBoosted)) {
+            uint preBoostedVotes = proposal.preBoostedVotes[YES] + proposal.preBoostedVotes[NO];
+            if (proposal.state == ProposalState.Closed) {
+              //give back reputation for the voter
+                rewards[1] = ((voter.reputation * params.votersReputationLossRatio)/100);
+            } else if (proposal.winningVote == voter.vote ) {
+                rewards[1] = (((voter.reputation * params.votersReputationLossRatio)/100) +
+                ((voter.reputation * lostReputation)/preBoostedVotes));
+            }
+            voter.reputation = 0;
+        }
+        //as proposer
+        if ((proposal.proposer == _beneficiary)&&(proposal.winningVote == YES)&&(proposal.proposer != address(0))) {
+            rewards[2] = params.proposingRepRewardConstA;
+            proposal.proposer = 0;
+        }
+        if (rewards[0] != 0) {
+            proposal.totalStakes = proposal.totalStakes.sub(rewards[0]);
+            require(stakingToken.transfer(_beneficiary, rewards[0]));
+            emit Redeem(_proposalId,organizations[proposal.organizationId],_beneficiary,rewards[0]);
+        }
+        if (rewards[1] + rewards[2] != 0 ) {
+            VotingMachineCallbacksInterface(proposal.callbacks).mintReputation(rewards[1] + rewards[2],_beneficiary,_proposalId);
+            emit RedeemReputation(_proposalId,organizations[proposal.organizationId],_beneficiary,rewards[1] + rewards[2]);
+        }
```

Comment:
should `return rewards`?

Reply:
rewards is returned automaticly as it defined as the returns var